share/functions/__fish_cached.fish

```diff
@@ -0,0 +1,21 @@
+function __fish_cached --description "Cache the command output for a given amount of time"
```

Comment:
Here I kept the behavior of hanging up the execution until the cache load command is finished.

This can still be improved. Maybe hanging up only when there is no cache file, and after that when the cache is expired, instantly return the cached response and an cache update command in background